### PR TITLE
Nullable annotate InvisibleEditor.cs

### PR DESCRIPTION
**Pull request by jasonmalinowski**

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/FindReferences/VisualStudioDefinitionsAndReferencesFactory.cs [hunk 1]
```diff
@@ -66,10 +66,9 @@ public VisualStudioDefinitionsAndReferencesFactory(SVsServiceProvider servicePro
         private string GetSourceLine(string filePath, int lineNumber)
         {
             using var invisibleEditor = new InvisibleEditor(
-                _serviceProvider, filePath, hierarchyOpt: null, needsSave: false, needsUndoDisabled: false);
+                _serviceProvider, filePath, hierarchy: null, needsSave: false, needsUndoDisabled: false);
             var vsTextLines = invisibleEditor.VsTextLines;
-            if (vsTextLines != null &&
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/FindReferences/VisualStudioDefinitionsAndReferencesFactory.cs:71]: VsTextLines has been revised to never be null.

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs [hunk 2]
```diff
@@ -60,14 +62,13 @@ public InvisibleEditor(IServiceProvider serviceProvider, string filePath, IVsHie
                 try
                 {
                     var docData = Marshal.GetObjectForIUnknown(docDataPtr);
-                    _vsTextLines = docData as IVsTextLines;
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs:63]: It's not clear to me why this was only using an 'as' cast to convert the docData to an IVsTextLines. We had a hard cast to IVsTextBuffer a line later, and although it's _possible_ that somebody could implement IVsTextBuffer but not IVsTextLines, I can't think of any reasonable scenario for that. Worse off, the only objects here that can be passed to GetDocumentBuffer and actually work would be the editor implementations anyways which would implement all the interfaces.

I'm figuring this all just evolved this way for no good reason, so I'm cleaning it up.

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs [hunk 3]
```diff
@@ -88,18 +89,18 @@ public InvisibleEditor(IServiceProvider serviceProvider, string filePath, IVsHie
             }
         }
 
-        private IVsProject TryGetProjectOfHierarchy(IVsHierarchy hierarchyOpt)
+        private IVsProject? TryGetProjectOfHierarchy(IVsHierarchy? hierarchy)
         {
             // The invisible editor manager will fail in cases where the IVsProject passed to it is not consistent with
             // the IVsProject known to IVsSolution (e.g. if the object is a wrapper like AbstractHostObject created by
             // the CPS-based project system). This method returns an IVsProject instance known to the solution, or null
             // if the project could not be determined.
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs:97]: @davkean The AbtractHostObject stuff would all have been the fake IVsHierarchies you used to create for each target; that's all gone now right? I imagine we can delete this hack entirely.

File: src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs [hunk 4]
```diff
@@ -82,6 +82,10 @@ internal override IInvisibleEditor OpenInvisibleEditor(DocumentId documentId)
                 }
             }
 
+            // Documents in the VisualStudioWorkspace always have file paths since that's how we get files given
+            // to us from the project system.
+            Contract.ThrowIfNull(textDocument.FilePath);
```
Comment (sharwell) [inline src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs:87]: ❔ Where would the previous implementation have thrown if this invariant was violated?

File: src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs [hunk 5]
```diff
@@ -491,7 +491,7 @@ protected override void ApplyDocumentTextChanged(DocumentId documentId, SourceTe
                 {
                     // The edits would get sent by the co-authoring service to the owner.
                     // The invisible editor saves the file on being disposed, which should get reflected  on the owner's side.
-                    using (var invisibleEditor = new InvisibleEditor(_serviceProvider, document.FilePath, hierarchyOpt: null,
+                    using (var invisibleEditor = new InvisibleEditor(_serviceProvider, document.FilePath!, hierarchy: null,
```
Comment (sharwell) [inline src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs:494]: ❔ How do we know `FilePath` is not null here? Is there already validation of this condition earlier in the code path?

File: src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs (hunk 4 above)
Comment (jasonmalinowski) [reply src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs:87]: Calling the VS API with an empty file path wouldn't have worked, that much is certain.

File: src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs (hunk 5 above)
Comment (jasonmalinowski) [reply src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs:494]: Ditto: code never could have worked prior. And more practically, remote files are backed by the file system in a certain magic temp directory.

**Pull request merged**